x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/capacity/AutoscalingDeciderResultsTests.java

```diff
@@ -57,21 +57,25 @@ public void testRequiredCapacity() {
         boolean node = randomBoolean();
         boolean storage = randomBoolean();
         boolean memory = randomBoolean() || storage == false;
+        boolean processor = randomBoolean() || memory == false;
```

Comment:
The intention here is to ensure at least one of storage, memory and processor has a value. I suppose that is still obtained now and we can still hit all valid combinations. But some are hit less likely than others I think. Can we do this instead:

```suggestion
        boolean memory = randomBoolean();
        boolean processor = randomBoolean() || (storage == false && memory == false);
```

Seems more logical/readable.